contracts/facets/Market/MarketFacet.sol

```diff
@@ -544,56 +542,68 @@ contract MarketFacet is IMarket {
 
     function endAuction(
         uint256 _tokenID,
+        address _tokenAddress,
         address _owner,
         address _creator
     ) external override onlyMediaCaller returns (bool) {
         LibMarketStorage.MarketStorage storage ms = LibMarketStorage.marketStorage();
         require(
-            uint256(ms._tokenAsks[_tokenID]._firstBidTime) != 0,
+            uint256(ms._tokenAsks[_tokenAddress][_owner][_tokenID]._firstBidTime) != 0,
             "Market: Auction hasn't begun"
         );
         require(
             block.timestamp >=
-                (ms._tokenAsks[_tokenID]._firstBidTime - ms._tokenAsks[_tokenID]._createdAt)
-                    + ms._tokenAsks[_tokenID]._duration,
+                (ms._tokenAsks[_tokenAddress][_owner][_tokenID]._firstBidTime - ms._tokenAsks[_tokenAddress][_owner][_tokenID]._createdAt)
```

Comment:
Why do you need to take into consideration `_firstBidTime`? Can we not just compare block time against `_createdAt + _duration`?

Reply:
because, auction does not start until first bid is placed so need to consider it for accurate calculation